worker/src/main/java/gov/cms/ab2d/worker/service/WorkerService.java

```diff
@@ -0,0 +1,65 @@
+package gov.cms.ab2d.worker.service;
+
+import gov.cms.ab2d.common.model.Job;
+import gov.cms.ab2d.common.repository.JobRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+import static gov.cms.ab2d.common.model.JobStatus.IN_PROGRESS;
+import static gov.cms.ab2d.common.model.JobStatus.SUCCESSFUL;
+
+/**
+ * This class is responsible for actually processing the job and preparing bulk downloads for users.
+ */
+@Slf4j
+@Service
+public class WorkerService {
+
+    private final JobRepository jobRepository;
+
+    public WorkerService(JobRepository jobRepository) {
+        this.jobRepository = jobRepository;
+    }
+
+    public void process(String jobId) throws IOException {
+
+        final Job job = jobRepository.findByJobUuid(jobId);
```

Comment:
Even though the `process` method itself should not be transactional as we discussed (it's long-running), _ some parts_ of the workflow still should. `process` begins by doing 3 things: find the job, verify it is still in `SUBMITTED` status, move to `IN_PROGRESS`. These 3 things should be done in a transaction; so I guess they need to be extracted into a separate _public_ method with tx annotation applied.

Reply:
This was a good discussion - helped me understand the scenario better.
As discussed, I have extracted a `JobProcessingService`  where I created a public method `putJobInProgress` annotated as follows:
```@Transactional(propagation = Propagation.REQUIRES_NEW, isolation = Isolation.SERIALIZABLE)```